src/components/HeaderWithLanguage/CloneImageDialog.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2025-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Formik, useFormikContext } from "formik";
+import { useCallback, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { DialogOpenChangeDetails } from "@ark-ui/react";
+import {
+  DialogContent,
+  Button,
+  DialogBody,
+  DialogHeader,
+  DialogRoot,
+  DialogTitle,
+  DialogTrigger,
+  Text,
+} from "@ndla/primitives";
+import { styled } from "@ndla/styled-system/jsx";
+import { ImageUploadFormElement } from "../../containers/ImageUploader/components/ImageUploadFormElement";
+import { useMessages } from "../../containers/Messages/MessagesProvider";
+import * as imageApi from "../../modules/image/imageApi";
+import { NdlaErrorPayload } from "../../util/resolveJsonOrRejectWithError";
+import { toEditImage } from "../../util/routeHelpers";
+import { DialogCloseButton } from "../DialogCloseButton";
+import { FormActionsContainer, FormContent } from "../FormikForm";
+import validateFormik from "../formikValidationSchema";
+
+interface Props {
+  loading: boolean;
+  imageId: number;
+}
+
+const StyledFormContent = styled(FormContent, {
+  base: {
+    marginInlineEnd: "large",
+    gap: "small",
+  },
+});
+
+interface FormikValuesType {
+  imageFile: Blob | string | undefined;
+}
+
+const formikRules = { imageFile: { required: true } };
+
+export const CloneImageDialog = ({ loading, imageId }: Props) => {
+  const { t, i18n } = useTranslation();
+  const [open, setOpen] = useState(false);
+  const { createMessage, applicationError } = useMessages();
+  const [updating, setUpdating] = useState(false);
+  const navigate = useNavigate();
+  const parentFormikContext = useFormikContext<unknown>();
+  const parentFormIsDirty = parentFormikContext.dirty;
+
+  const onOpenChange = useCallback((details: DialogOpenChangeDetails) => setOpen(details.open), []);
+  const handleSubmit = useCallback(
+    async (values: FormikValuesType) => {
+      try {
+        if (parentFormIsDirty) {
+          createMessage({
+            translationKey: "form.mustSaveFirst",
+            severity: "danger",
+            timeToLive: 0,
+          });
+        } else {
+          setUpdating(true);
+          const newImage = await imageApi.cloneImage(imageId, values.imageFile);
+          navigate(toEditImage(newImage.id, newImage.title.language));
```

Comment:
Må ikke updating settes til false her? er dette noe som vi kunne latt react-query håndtere for oss?

Reply:
Vi navigerer ut av komponenten uansett vel?
Kan det ikke bli noe krøll med å forsøke å oppdatere state på en unmounted komponent da?

Reply:
Det er sant! men hadde det vært bedre å basere oss på loading-state fra react-query? så trenger vi ikke håndtere det selv

Reply:
Ouf, jeg misliker react-query, men jeg folder meg naturligvis etter de som faktisk vedlikeholder kodebasens ønsker :nerd_face: :mage_woman: